xenon/core.py

```diff
@@ -34,6 +36,24 @@ def analyze(args, logger):
     return find_infractions(args, logger, results), results
 
 
+def build_blocks_to_ignore(args):
+    '''Build a dict from module name to a list of blocks in that module to ignore.
+
+    Read the data from args.ignore_blocks which is a list of blocks separated by ",".
+    The format of a block is "<module_name>:<block_name>".
+    '''
+    if not args.ignore_blocks:
+        return {}
+
+    blocks_to_ignore = defaultdict(list)
+
+    for block_to_ignore in args.ignore_blocks.split(','):
+        module_name, block_name = block_to_ignore.split(':')
+        blocks_to_ignore[module_name.strip()].append(block_name.strip())
```

Comment:
Let's use `defaultdict(set)` and `.add`, since this is used to check for existence. For big codebases it'll make a difference as membership checks with set are an O(1) operation as opposed to O(n) with lists.